models/nlp/qa/huggingface/inference.py

```diff
@@ -0,0 +1,83 @@
+import csv
+import zipfile
+
+import torch
+import valohai
+from transformers import AutoModelForQuestionAnswering, AutoTokenizer
+
+from utils.serializers import get_serializer
+
+
+def load_model(model_path):
+    tokenizer = AutoTokenizer.from_pretrained(model_path)
+    model = AutoModelForQuestionAnswering.from_pretrained(model_path)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model.to(device)
+    return tokenizer, model, device
+
+
+def load_data(data_path):
+    data = []
+    with open(data_path) as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            data.append(
+                {
+                    "id": row["id"],
+                    "context": row["context"],
+                    "question": row["question"],
+                },
+            )
+    return data
+
+
+def predict(question, tokenizer, model, device):
+    inputs = tokenizer(question["question"], question["context"], return_tensors="pt")
+    inputs = {key: tensor.to(device) for key, tensor in inputs.items()}
+    with torch.no_grad():
+        outputs = model(**inputs)
+    answer_start_index = outputs.start_logits.argmax()
+    answer_end_index = outputs.end_logits.argmax()
+    predict_answer_tokens = inputs["input_ids"][
+        0,
+        answer_start_index : answer_end_index + 1,
+    ].cpu()
+    return tokenizer.decode(predict_answer_tokens)
+
+
+def main():
+    model_path = valohai.inputs("model").path(process_archives=False)
+    if model_path.endswith(".zip"):
+        with zipfile.ZipFile(model_path, "r") as zip_ref:
+            zip_ref.extractall("/tmp/model")  # noqa: S108
+    else:
+        print(f"Model {model_path} is not a zip file. Aborting.")
+        return
+
+    tokenizer, model, device = load_model("/tmp/model")  # noqa: S108
+    data_path = valohai.inputs("data").path("*.csv")
+    rows = load_data(data_path)
+
+    results = []
+    for index, row in enumerate(rows):
+        answer = predict(row, tokenizer, model, device)
+        results.append(
+            {
+                "id": row["id"],
+                "context": row["context"],
+                "question": row["question"],
+                "answer": answer,
+            },
+        )
```

Comment:
You're already in control of the format of `row`, no need to repeat yourself.
```suggestion
        results.append({**row, "answer": answer})
```